plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionFields.java

```diff
@@ -29,7 +29,7 @@
 
 public final class PartitionFields
 {
-    private static final String NAME = "[a-z_][a-z0-9_]*";
+    private static final String NAME = "[a-z_\\.?][a-z0-9_\\.?]*";
```

Comment:
This should be more strict. For example, shouldn't match `a...b`.

```java
IDENTIFIER = "[a-z_][a-z0-9_]*";
NAME = IDENTIFIER + "(\\." + NAME + ")*";
```

also, since we thus allow partitioning on a function from a field (like `year(rec.timestamp)`), we should add some test coverage for such usage too